client/client.go

```diff
@@ -38,20 +41,21 @@ func lookupSrv(name string) ([]string, error) {
 	}
 	_, srvs, err := net.LookupSRV("", "", name)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %s", errNotAnSRV, err)
```

Comment:
Usually the "cause" is wrapped and goes after the colon. Did you mean `%v: %w`?

Reply:
An error here does not always mean it is "not an SRV".

Reply:
The error means we don't need to apply SRV resolving logic here because that's not an SRV for sure.

Reply:
I see how it is intended by you, but it is the "for sure" part that I disagree with.

Reply:
The failure can be caused by "not an SRV", or by a network timeout during the DNS communications, or a buggy DNS server sending malformed response, or something else.

Reply:
I see why are you arguing.

I didn't mean to say that  `net.LookupSRV()` is a method to check is it an SRV or not.
What I've tried to say, is that we have SRV and not SRV logic. My PR continues the best-effort way to resolve the record.
And `errNotAnSRV` helps to switch logic (further resolving or using as-is).